.cirrus.yml

```diff
@@ -0,0 +1,16 @@
+freebsd_instance:
+  image: freebsd-12-0-release-amd64
+freebsd_test_task:
+  env:
+    CC: "clang70"
```

Comment:
> are you overriding the default system C/C++ compiler here?

Yes, this is overriding the use of `/usr/bin/cc` and `/usr/bin/c++` with Clang from the LLVM 7.0 port, since building requires the LLVM files, which are not available by default. I figured it made sense for the version of the compiler to match the version of LLVM.

> only the system one is correct, when depending on packaged llvm.

Oh interesting, okay. So does the version of the packaged LLVM need to match version of the default compiler? That's 6.0 in FreeBSD 12.0.

> I think it would be better to leave these env vars alone.

:+1:

Reply:
It makes sense why you would want to match the compiler with the library, but there's a different factor at play here, which is that C++ ABI is brittle and different compiler versions have incomptabilities. For many projects these are not noticed, but with LLVM and Clang libraries, there have been several ABI incompatibilities over the lifetime of the Zig project. The only surefire solution is to make sure Zig is built with the same compiler that the LLVM/Clang libraries are built with. So that's the only thing supported by Zig. That gives users 2 choices: 
 * Use the default system compiler + LLVM/Clang libraries from the package manager, or
 * Use any compiler, and build LLVM/Clang from source, and then use that to build Zig from source.

To be clear: there is no relationship between the LLVM 7.0.0 and Clang 7.0.0 libraries that Zig depends on, and the compiler that you use to build Zig. Zig needs exactly LLVM and Clang 7.0.0 libraries, but any C++ compiler can be used to build Zig, provided that same compiler was used to build the LLVM and Clang 7.0.0 libraries.

Reply:
I see, that definitely makes sense. Thanks!